elasticsearch/src/main/java/org/hibernate/search/elasticsearch/client/impl/ArbitrarySort.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.elasticsearch.client.impl;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.hibernate.search.elasticsearch.impl.GsonHolder;
+
+import io.searchbox.core.search.sort.Sort;
+
+/**
+ * Implementation of a sort with arbitrary keys and values for Jest.
+ * <p>This is useful for providing a way for users to use sort features
+ * that are not otherwise implemented in Hibernate Search.
+ *
+ * @author Yoann Rodiere
+ */
+public class ArbitrarySort extends Sort {
+
+	private String fieldName;
+
+	private Object sortDescription;
+
+	public ArbitrarySort(String fieldName, String sortDescription) {
+		super( fieldName );
```

Comment:
It's just easier to implement given that `org.hibernate.search.query.dsl.sort.impl.NativeSortField` is already split into 2 pieces. Upon constructing an `ArbitrarySort` I could have merged the field name and the JSON sort definition into a single JSON sort, but it seemed pointless. Especially since it's an implementation package and we're not offering `ArbitrarySort` to users as an API.

On the subject of `NativeSortField`, there was two reasons for splitting it, but not strong ones:

 1. Implementation reason: if we actually know the field name, it avoids providing the super `SortField` constructor with a dummy field name, which would make this class even more hackish than it is now.
 2. API reason: Elasticsearch seems to only use sorts in the form `'fieldName': <sortspec>`, so thought it would be easier for users to separate the target field from the sort spec.

Anyway, as I said these are not strong requirements. I can change it if you want.